test/spec/link.spec.jsx

```diff
@@ -2,38 +2,152 @@ import React from 'react';
 import { mount } from 'enzyme';
 
 import { PUSH, REPLACE } from 'src/action-types';
-import Link from 'src/link';
+
+import provideRouter from 'src/provider';
+import {
+  Link as RawLink,
+  PersistentQueryLink as RawPersistentQueryLink
+} from 'src/link';
+
+import bootstrap from './util/bootstrap';
+
+const { storeFactory, history } = bootstrap();
 
 describe('Router link component', () => {
-  it('dispatches a PUSH action with the correct href when clicked', done => {
-    const dispatch = action => {
-      expect(action).to.deep.equal({
-        type: PUSH,
-        payload: {
-          pathname: '/yo'
+  describe('Vanilla link', () => {
+    describe('PUSH', () => {
+      const hrefs = [
+        '/home/messages/a-team?test=ing',
+        {
+          pathname: '/home/messages/a-team?test=ing',
+          query: {
+            test: 'ing'
+          }
+        }
+      ];
+      hrefs.forEach(href =>
+        it('dispatches a PUSH action with the correct href when clicked', done => {
+          const assertion = (state, action) => {
+            if (action.type === PUSH) {
+              const { payload } = action;
+              expect(payload).to.have.property('pathname')
+                .that.contains('/home/messages/a-team');
+              expect(payload).to.have.property('query')
+                .that.deep.equals({ test: 'ing' });
+              done();
```

Comment:
```es6
const isNotLeftClick = e => e.button !== LEFT_MOUSE_BUTTON;
```

`e.button` is undefined in `enzyme` simulated clicks. This means that we never reach the real `<Link>` behavior. Here's the fix:

```es6
const isNotLeftClick = e => e.button && e.button !== LEFT_MOUSE_BUTTON;
```

😆